skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldLabels.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.skaffold
+
+import com.google.container.tools.core.PluginInfo
+import com.intellij.openapi.application.ApplicationInfo
+
+const val IDE_LABEL = "ide"
+const val IDE_VERSION_LABEL = "ideVersion"
+const val PLUGIN_VERSION_LABEL = "ijPluginVersion"
+
+/**
+ * Maintains list of Kubernetes labels - string based key-value pairs used by Skaffold to apply
+ * to all deployments. [getDefaultLabels] function supplies default set of labels used for all
+ * deployments.
+ */
+class SkaffoldLabels {
+    companion object {
+        /**
+         * Default set of labels for all Skaffold-based deployments. Includes IDE type and
+         * version, plugin version.
+         */
+        val defaultLabels: SkaffoldLabels by lazy {
+            SkaffoldLabels().apply {
+                this.labels[IDE_LABEL] = PluginInfo.platformPrefix
+                this.labels[IDE_VERSION_LABEL] = ApplicationInfo.getInstance().getStrictVersion()
+                this.labels[PLUGIN_VERSION_LABEL] = PluginInfo.pluginVersion
+            }
+        }
+    }
+
+    val labels: MutableMap<String, String> = mutableMapOf()
```

Comment:
this can be private no?

Also I wonder if there is a way to not initialize this to an empty map and indicate that its lazy?

Reply:
oh I guess it can't be private. hmm. My intention with the comment was to try to prevent instantiation of the class with just an empty map.

Reply:
We can still do this lazy I think, since otherwise we keep creating this map.

Reply:
someone could do this:

```
val skaffoldLabels = SkaffoldLabels()
println(skaffoldLabels.labels) // returns an empty map
```

I was just trying to avoid this since. maybe a private constructor?